src/rp2/plugin/report/us/open_positions.py

```diff
@@ -0,0 +1,233 @@
+        total_cost: RP2Decimal = RP2Decimal("0")
+        for asset, computed_data in asset_to_computed_data.items():
+            if not isinstance(asset, str):
+                raise RP2TypeError(f"Parameter 'asset' has non-string value {asset}")
+            ComputedData.type_check("computed_data", computed_data)
+
+            net_cost = RP2Decimal("0")
+            for current_transaction in computed_data.in_transaction_set:
+                in_xact = cast(InTransaction, current_transaction)
+                net_cost += in_xact.fiat_in_with_fee * (RP2Decimal("1") - computed_data.get_in_lot_sold_percentage(in_xact))
+
+            if net_cost < _FIAT_EMPTY_BAL_CUTOFF:
+                continue
+
+            total_cost += net_cost
+
+        # Loop thru every asset again, but this time we can do reporting.
+        for asset, computed_data in asset_to_computed_data.items():
+            if not isinstance(asset, str):
+                raise RP2TypeError(f"Parameter 'asset' has non-string value {asset}")
+            ComputedData.type_check("computed_data", computed_data)
+
+            # Net cost basis will be the sum of all usd_in_w_fee for an asset after multiplying off the sold percentage.
+            net_cost = RP2Decimal("0")
+            for current_transaction in computed_data.in_transaction_set:
+                in_xact = cast(InTransaction, current_transaction)
+                sold_pct: RP2Decimal = computed_data.get_in_lot_sold_percentage(in_xact)
+                net_cost += in_xact.fiat_in_with_fee * (RP2Decimal("1") - sold_pct)
+
+            # Sum the total amount of crypto asset remaining.
+            tot_crypto_bal = RP2Decimal("0")
+            for balset in computed_data.balance_set:
+                tot_crypto_bal += balset.final_balance
+
+            if net_cost < _FIAT_EMPTY_BAL_CUTOFF:
+                continue
+            # print("asset ",asset,tot_crypto_bal,net_cost,float(net_cost))
+
+            # For report clarity, this cuts down the decimals in the unit price based on the value of the unt, e.g. $1+
+            # shows price to the nearest cent while $0.20-$1.00 shows to 4 decimal places, and anything smaller is
+            # left un-rounded (the template formatting currently would show up to 7 decimal places.)
+            unit_cost_basis: RP2Decimal = net_cost / tot_crypto_bal
+            if unit_cost_basis >= _FIAT_UNIT_WINDOW_2_DECIMAL:
+                unit_cost_basis = RP2Decimal(unit_cost_basis.quantize(Decimal("1.00")))
+            elif unit_cost_basis >= _FIAT_UNIT_WINDOW_4_DECIMAL:
+                unit_cost_basis = RP2Decimal(unit_cost_basis.quantize(Decimal("1.0000")))
+
+            # Only 1 record in the AP sheet per asset.
+            ap_sheet.append_rows(1)
+            ap_row_index: int = row_indexes["AssetPrice"]
+            self._fill_cell(ap_sheet, ap_row_index, 0, asset)
+            self._fill_cell(ap_sheet, ap_row_index, 1, 0)
```

Comment:
Since there's a VLOOKUP on the two report tabs, I don't want to inadvertently instruct the user to key in prices directly to the report tabs. Instead I've chosen to use "See Input Tab" to direct their attention there.

Reply:
Oh, my bad. I meant: add the "enter price" string in the input tab.